hw_diag/diagnostics/ecc_diagnostic.py

```diff
@@ -21,5 +22,5 @@ def perform_test(self, diagnostics_report):
                       str(json.dumps(ecc_tests))
                 diagnostics_report.record_failure(msg, self)
 
-        except Exception as e:
+        except EccMalfunctionException as e:
```

Comment:
You are missing some other possible execptions:
- [GatewayMFRFileNotFoundException](https://github.com/NebraLtd/hm-pyhelper/blob/a98ff5e135fda14276239f66e6545d857af41362/hm_pyhelper/miner_param.py#L51)
- [ResourceBusyError](https://github.com/NebraLtd/hm-pyhelper/blob/a98ff5e135fda14276239f66e6545d857af41362/hm_pyhelper/lock_singleton.py#L66)

And [KeyDiagnostics](https://github.com/NebraLtd/hm-diag/blob/12420cfb08880a0b296b4130174f8563c8074ea1/hw_diag/diagnostics/key_diagnostics.py#L27) is an additional class that will need these checks. Should add unit tests for handling each exception.